src/Api/Controllers/OrganizationsController.cs

```diff
@@ -139,7 +142,10 @@ public async Task<OrganizationSubscriptionResponseModel> GetSubscription(string
                 throw new NotFoundException();
             }
 
-            return new OrganizationSubscriptionResponseModel(organization, subscriptionInfo);
+            var provider = await _providerRepository.GetByOrganizationIdAsync(orgIdGuid);
+            var hideSensitiveData = provider != null && !_currentContext.ProviderUser(provider.Id);
+
+            return new OrganizationSubscriptionResponseModel(organization, subscriptionInfo, hideSensitiveData);
```

Comment:
I think we should create the same permissions on the server as we did in clients, i.e. `ViewSubscription`, `EditSubscription`, `EditPaymentMethods`, `ViewBillingHistory`. These will be on `CurrentContext` instead of `ManageBilling`. Then we can just use those to determine what data to include here (and to block off any endpoints per my comment above). And it keeps clients & server consistent.

Reply:
That is smart, I've implemented it. Let me know what you think.